### PR TITLE
fix: check if anchor commit exists before anchoring CDB-1985

### PR DESCRIPTION
**Pull request by stephhuynh18**

### CODE REVIEW

**Review by stbrody — COMMENTED** [review]
will this result in fixing the database to actually mark those requests as anchored with the right anchor commit CID?

**Comment by stephhuynh18** [comment]
> will this result in fixing the database to actually mark those requests as anchored with the right anchor commit CID?

We spoke about this in other channels but for documentation: 
Since some of our requests are actually COMPLETED but their status is still READY or PROCESSING or PENDING, this checks to see if there is an anchor commit associated with the request. If so it marks the candidate stream as already anchored. 
This results in the request being updated to COMPLETED, and prevents us from re-anchoring it.

File: src/repositories/anchor-repository.ts [hunk 1]
```diff
@@ -30,10 +30,13 @@ export class AnchorRepository {
...
-    const { requestId, ...anchorWithoutRequestId } = await connection(TABLE_NAME)
-      .where({ requestId: request.id })
-      .first()
+    const results = await connection(TABLE_NAME).where({ requestId: request.id }).first()
 
+    if (!results) {
+      return results
+    }
+
+    const { requestId, ...anchorWithoutRequestId } = results
```
[the remark this answers is not in the document]
Comment (stephhuynh18) [reply src/repositories/anchor-repository.ts:39]: actually i'm not even going to bother deleting it. The types being used are inaccurate and i shall update to accurately reflect what is being returned

**Review by smrz2001 — APPROVED** [review]
:shipit:

**Pull request merged**